### PR TITLE
Monero wallet cleanup

### PR DESCRIPTION
**Pull request by da-kami**

### CODE REVIEW

File: swap/src/protocol/alice/steps.rs [hunk 1]
```diff
@@ -49,15 +48,12 @@ pub async fn wait_for_locked_bitcoin(
...
+pub async fn lock_xmr(
     bob_peer_id: PeerId,
     state3: alice::State3,
     event_loop_handle: &mut EventLoopHandle,
-    monero_wallet: Arc<W>,
-) -> Result<()>
-where
-    W: Transfer,
-{
+    monero_wallet: Arc<monero::Wallet>,
```
Comment (thomaseizinger) [inline swap/src/protocol/alice/steps.rs:55]: Could this be a `&monero::Wallet`?

File: swap/src/monero/wallet.rs [hunk 2]
```diff
@@ -40,79 +38,31 @@ impl Wallet {
...
+            self.inner.lock().await.create_wallet(self.name.as_str()).await.context(
+                "Unable to create Monero wallet, please ensure that the monero-wallet-rpc is available",
+            )?;
 
-        let tx_hashes = sweep_all.tx_hash_list.into_iter().map(TxHash).collect();
-        Ok(tx_hashes)
-    }
+            info!("Created Monero wallet {}", self.name);
+        } else {
+            info!("Opened Monero wallet {}", self.name);
```
Comment (thomaseizinger) [inline swap/src/monero/wallet.rs:50]: Make these `debug` please.

**Review by thomaseizinger — APPROVED** [review]

**Review by rishflab — APPROVED** [review]

**Comment by da-kami** [comment]
@thomaseizinger @rishflab  I broke the re-ordering of code blocks in the wallet into smaller chunks on top of removing the traits. You can have another look if you want, should be easier to review now :)

**Pull request merged**